account_document_reversal/models/account_bank_statement.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class AccountPayment(models.Model):
+    _name = 'account.bank.statement.line'
+    _inherit = ['account.bank.statement.line', 'account.document.reversal']
+
+    @api.multi
+    def button_cancel_reconciliation(self):
+        """ If cancel method is to reverse, use document reversal wizard """
+        cancel_reversal = all(self.mapped('journal_entry_ids.move_id.'
+                                          'journal_id.is_cancel_reversal'))
+        states = self.mapped('statement_id.state')
+        if cancel_reversal:
+            if not all(st == 'open' for st in states):
+                raise UserError(
+                    _('Only new bank statement can be cancelled'))
+            return self.reverse_document_wizard()
+        return super().button_cancel_reconciliation()
+
+    @api.multi
+    def action_document_reversal(self, date=None, journal_id=None):
+        """ Reverse all moves related to this statement + delete payment """
+        # This part is from button_cancel_reconciliation()
+        aml_to_unbind = self.env['account.move.line']
```

Comment:
@jbeficent  here both way are working same but Duplicate Variable Declaration
So, trying to ` aml_to_unbind = aml_to_cancel = self.env['account.move.line']`

Same as a for `self.env['account.payment']`